seisbench/generate/augmentation.py

```diff
@@ -101,6 +101,28 @@ def __str__(self):
         return f"Normalize ({desc})"
 
 
+class Copy:
+    """
+    A copy augmentation. Maps data from a given key in the state_dict to a new key.
+
+    :param key: The keys for reading from and writing to the state dict.
+                A a 2-tuple is expected, with the first string indicating the key to
+                read from and the second one the key to write to.
+    :type key: str, tuple[str, str]
+
+    """
+
+    def __init__(self, key=("X", "Xc")):
+        self.key = key
+
+    def __call__(self, state_dict):
+        x, metadata = state_dict[self.key[0]]
+        state_dict[self.key[1]] = (x, metadata)
```

Comment:
I would suggest to make deep instead of shallow copies here, i.e., change the lower line to `... = (x.copy(), copy.deepcopy(metadata))`.

Reply:
Good point, done and added test to check it is working as expected.